splice_machine/src/main/java/com/splicemachine/derby/stream/utils/ExternalTableUtils.java

```diff
@@ -0,0 +1,204 @@
+
+        boolean prepared = false;
+        SpliceTransactionResourceImpl transactionResource = null;
+        Txn txn = null;
+        LanguageConnectionContext lcc = null;
+        try {
+            if (activation == null) {
+                txn = SIDriver.driver().lifecycleManager()
+                        .beginTransaction();
+                transactionResource = new SpliceTransactionResourceImpl();
+                prepared=transactionResource.marshallTransaction(txn);
+                lcc = transactionResource.getLcc();
+            }
+            else {
+                lcc = activation.getLanguageConnectionContext();
+            }
+            DataDictionary dd = lcc.getDataDictionary();
+            ConglomerateDescriptor cd = dd.getConglomerateDescriptor(conglomerateId);
+            UUID tableId = cd.getTableID();
+            TableDescriptor td = dd.getTableDescriptor(tableId);
+            ColumnDescriptorList cdl = td.getColumnDescriptorList();
+            ExecRow execRow = new ValueRow(cdl.size());
+            DataValueDescriptor[] dvds = execRow.getRowArray();
+            for (int i = 0; i < cdl.size(); ++i) {
+                ColumnDescriptor columnDescriptor = cdl.get(i);
+                DataTypeDescriptor dtd = columnDescriptor.getType();
+                dvds[i] = dtd.getNull();
+            }
+            StructType schema = execRow.schema();
+            return  schema;
+        }
+        catch (Exception e) {
+            throw StandardException.plainWrapException(e);
+        }
+        finally {
+            if(prepared)
+                transactionResource.close();
+            try {
+                if (txn != null)
+                    txn.commit();
+            } catch (IOException ioe) {
+                throw StandardException.plainWrapException(ioe);
+            }
+        }
+    }
+
+    public static void checkSchema(StructType tableSchema,
+                                   StructType dataSchema,
+                                   int[] partitionColumnMap,
+                                   String location) throws StandardException{
+
+
+        StructField[] tableFields = tableSchema.fields();
+        StructField[] dataFields = dataSchema.fields();
+
+        if (tableFields.length != dataFields.length) {
+            throw StandardException.newException(SQLState.INCONSISTENT_NUMBER_OF_ATTRIBUTE,
+                    tableFields.length, dataFields.length, location);
+        }
+
+        StructField[] partitionedTableFields = new StructField[tableSchema.fields().length];
+        Set<Integer> partitionColumns = new HashSet<>();
+        for (int pos : partitionColumnMap) {
+            partitionColumns.add(pos);
+        }
+        int index = 0;
+        for (int i = 0; i < tableFields.length; ++i) {
+            if (!partitionColumns.contains(i)) {
+                partitionedTableFields[index++] = tableFields[i];
+            }
+        }
+        for (int pos:partitionColumnMap) {
```

Comment:
It seems that we don't need to populate the partition columns here, as we are not check the consistency of partitioning columns between data schema and table schema.

Reply:
Yes